tempodb/backend/azure/azure_helpers.go

```diff
@@ -20,7 +21,22 @@ const (
 )
 
 func GetContainerURL(ctx context.Context, cfg *Config, hedge bool) (blob.ContainerURL, error) {
-	c, err := blob.NewSharedKeyCredential(cfg.StorageAccountName.String(), cfg.StorageAccountKey.String())
+	accountName := cfg.StorageAccountName.String()
+	accountKey := cfg.StorageAccountKey.String()
+
+	if accountName == "" {
+		accountName = os.Getenv("AZURE_STORAGE_ACCOUNT")
+	}
+
+	if accountKey == "" {
+		accountKey = os.Getenv("AZURE_STORAGE_KEY")
+	}
+
+	if accountKey == "" || accountName == "" {
```

Comment:
remove this check. let's allow `blob.NewSharedKeyCredential()` to return an error for this. There may be times where leaving one or both of these blank is fine.